CMakeLists.txt

```diff
@@ -23,7 +23,24 @@ include( CTest )
 
 #---------------------------------------------------------------------
 # Find ITK.
-find_package( ITK 5.1.1 REQUIRED )
+find_package( ITK 5.1.1 REQUIRED COMPONENTS
+    ITKCommon
+    ITKDistanceMap
+    ITKImageGrid
+    ITKImageIntensity
+    ITKIOImageBase
+    ITKIOMeshBase
+    ITKIOMeshOBJ
+    ITKIOMeta
+    ITKIOXML
+    ITKMathematicalMorphology
+    ITKOptimizers
+    ITKRegistrationCommon
+    ITKSmoothing
+    ITKTestKernel
+    ITKThresholding
+    ITKTransform
```

Comment:
Thanks @dzenanz A question: are these _all_ ITK components? Or are these exactly the ones needed by elastix?

Reply:
These are the components required by elastix to compile for me. ITK has around 100 components total. Some components will be included indirectly.

List of supported IO formats should be specified explicitly now. I only included .obj mesh format (ITKIOMeshOBJ) and .mha/.mhd image format (ITKIOMeta). You will want to expand that to include all of the [IO modules](https://github.com/InsightSoftwareConsortium/ITK/tree/master/Modules/IO), or at least a significant subset.

I will amend the commit to optionally include ITK GPU module(s).

Reply:
Was the [`WhatModulesITK.py`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/Maintenance/WhatModulesITK.py) script used to determine these? Modules could be missing.

Also, for the elastix CLI, we should include all the IO modules by including `ITKImageIO`, `ITKMeshIO`, and `ITKTransformIO`.

Reply:
I wrote another script to determine the ITK modules elastix is using

Reply:
@thewtex I can't find the modules you mentioned?
The closest I got was: ITKIOImageBase and ITKIOTransformBase (which are already included).
I did find ITKIOMesh, which I'll include.

Reply:
Those are meta-modules. First introduced with [4.13](https://github.com/InsightSoftwareConsortium/ITK/blob/9c0cdee0b6052da4840d3cb33d20a253b3cccde9/Documentation/ReleaseNotes/4.13.md#introduction).

Reply:
Yes, but I can't find these modules (ITKImageIO and IKTTransformIO) on https://itk.org/Doxygen/html/modules.html which should list all available modules, right?

Reply:
I added them.